internal/elasticsearch/security/role_mapping/data_source.go

```diff
@@ -0,0 +1,112 @@
+package role_mapping
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/elastic/terraform-provider-elasticstack/internal/clients"
+	providerschema "github.com/elastic/terraform-provider-elasticstack/internal/schema"
+	"github.com/elastic/terraform-provider-elasticstack/internal/utils"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func NewRoleMappingDataSource() datasource.DataSource {
+	return &roleMappingDataSource{}
+}
+
+type roleMappingDataSource struct {
+	client *clients.ApiClient
+}
+
+func (d *roleMappingDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_elasticsearch_security_role_mapping"
+}
+
+func (d *roleMappingDataSource) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "Retrieves role mappings. See, https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-get-role-mapping.html",
+		Blocks: map[string]schema.Block{
+			"elasticsearch_connection": providerschema.GetEsFWConnectionBlock("elasticsearch_connection", false),
+		},
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				MarkdownDescription: "Internal identifier of the resource",
+				Computed:            true,
+			},
+			"name": schema.StringAttribute{
+				MarkdownDescription: "The distinct name that identifies the role mapping, used solely as an identifier.",
+				Required:            true,
+			},
+			"enabled": schema.BoolAttribute{
+				MarkdownDescription: "Mappings that have `enabled` set to `false` are ignored when role mapping is performed.",
+				Computed:            true,
+			},
+			"rules": schema.StringAttribute{
+				MarkdownDescription: "The rules that determine which users should be matched by the mapping. A rule is a logical condition that is expressed by using a JSON DSL.",
+				Computed:            true,
```

Comment:
```suggestion
				Computed:            true,
				CustomType: jsontypes.NormalizedType{}
```

We'll need to use the `NormalizedType` within the data source as well since we're re-using the model. This change also applies to `role_templates` and `metadata` attributes as well.